packages/query-core/src/utils.ts

```diff
@@ -361,7 +362,9 @@ function hasObjectPrototype(o: any): boolean {
 
 export function sleep(timeout: number): Promise<void> {
   return new Promise((resolve) => {
-    setTimeout(resolve, timeout)
+    const setTimeoutFn =
+      timeout === 0 ? systemSetTimeoutZero : managedSetTimeout
```

Comment:
why do we need to handle `setTimeout(0)` differently?

Reply:
Changed to managedSetTimeout